File: benchmarks/parse-json.pl
use Time::HiRes qw(time);
use JSON::XS;

$/ = '';
my $tt = time;

my $t = time;
open JSON, 'example.json' or die "Can't open file: $!\n";
my $json = <JSON>;
close JSON;
printf "json.read: %dms\n", int((time - $t) * 1000 + 0.5);

my $obj;
$t = time;
$obj = JSON::XS->new->ascii->decode($json) foreach 1 .. 100;
printf "json.parse100: %dms\n", int((time - $t) * 1000 + 0.5);

$t = time;
map { my %new_obj; $m = $_; $new_obj{$_} = $m{$_} foreach keys %$m; } @{$obj->{'data'}->{'messages'}};
printf "json.transform: %dms\n", int((time - $t) * 1000 + 0.5);

printf "all: %dms\n", int((time - $tt) * 1000 + 0.5);

File: benchmarks/parse-json.js
var fs = require("fs");

console.time('all');
console.time('json.read');


fs.readFile('example.json', function (err, json){
	console.timeEnd('json.read');
	console.time('json.parse');

	var messages = JSON.parse(json).data.messages;

	console.timeEnd('json.parse');
	console.time('json.transform');

	var clone = messages.map(function (msg){
		var obj = {};
		for( var key in msg ){
			obj[key] = msg[key];
		}
		return obj;
	});

	console.timeEnd('json.transform');
	console.timeEnd('all');
});
